module/metrics/network.go

```diff
@@ -16,128 +16,148 @@ const (
 )
 
 type NetworkCollector struct {
-	outboundMessageSize             *prometheus.HistogramVec
-	inboundMessageSize              *prometheus.HistogramVec
-	duplicateMessagesDropped        *prometheus.CounterVec
-	queueSize                       *prometheus.GaugeVec
-	queueDuration                   *prometheus.HistogramVec
-	inboundProcessTime              *prometheus.CounterVec
-	outboundConnectionCount         prometheus.Gauge
-	inboundConnectionCount          prometheus.Gauge
-	dnsLookupDuration               prometheus.Histogram
-	dnsCacheMissCount               prometheus.Counter
-	dnsCacheHitCount                prometheus.Counter
-	dnsCacheInvalidationCount       prometheus.Counter
-	unstakedOutboundConnectionCount prometheus.Gauge
-	unstakedInboundConnectionCount  prometheus.Gauge
-}
-
-func NewNetworkCollector() *NetworkCollector {
-
-	nc := &NetworkCollector{
-
-		outboundMessageSize: promauto.NewHistogramVec(prometheus.HistogramOpts{
+	outboundMessageSize       *prometheus.HistogramVec
+	inboundMessageSize        *prometheus.HistogramVec
+	duplicateMessagesDropped  *prometheus.CounterVec
+	queueSize                 *prometheus.GaugeVec
+	queueDuration             *prometheus.HistogramVec
+	inboundProcessTime        *prometheus.CounterVec
+	outboundConnectionCount   prometheus.Gauge
+	inboundConnectionCount    prometheus.Gauge
+	dnsLookupDuration         prometheus.Histogram
+	dnsCacheMissCount         prometheus.Counter
+	dnsCacheHitCount          prometheus.Counter
+	dnsCacheInvalidationCount prometheus.Counter
+
+	prefix string
+}
+
+type NetworkCollectorOpt func(*NetworkCollector)
+
+func WithNetworkPrefix(prefix string) NetworkCollectorOpt {
+	return func(nc *NetworkCollector) {
+		nc.prefix = prefix
+	}
+}
+
+func NewNetworkCollector(opts ...NetworkCollectorOpt) *NetworkCollector {
+	nc := &NetworkCollector{}
+
+	for _, opt := range opts {
+		opt(nc)
+	}
+
+	nc.outboundMessageSize = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
 			Namespace: namespaceNetwork,
 			Subsystem: subsystemGossip,
-			Name:      "outbound_message_size_bytes",
+			Name:      nc.prefix + "outbound_message_size_bytes",
```

Comment:
Highlighting this comment.

Reply:
Thanks, this is good catch. 

I think since staked / unstaked network are using two completely separate libp2p hosts, it does make sense to have them as completely separate metrics. Furthermore, when @vishalchangrani and I originally talked about adding these unstaked metrics a while ago, we considered using tags, but in the end we decided against it because there were some downsides to using tags which I don't exactly remember now. @vishalchangrani?

2d60610